tests/trunc-double-uniform.ispc

```diff
@@ -0,0 +1,127 @@
+
+export void f_f(uniform float RET[], uniform float aFOO[]) {
+
+    uniform float error = 0;
+    RET[programIndex] = 0;
+    uniform double deltaLow = 0.000001d;
+    uniform double deltaHigh = 0.999999d;
+    uniform double testVal;
+    uniform double resVal;
+    uniform double baseVal;
+
+    //Case 1 : Generic value 1 < val < +inf with deltaLow.
+    baseVal = aFOO[programCount - 1];
+    testVal = baseVal + deltaLow;
+    resVal = trunc(testVal);
+    if (baseVal != resVal) {
+        error++;
+    }
+
+    //Case 2 : Generic value -inf < val < -1 with deltaLow.
+    baseVal = -aFOO[programCount - 1];
+    testVal = baseVal - deltaLow;
+    resVal = trunc(testVal);
+    if (baseVal != resVal) {
+        error++;
+    }
+
+    //Case 3 : Generic value 1 < val < +inf with deltaHigh.
+    baseVal = aFOO[programCount - 1];
+    testVal = baseVal + deltaHigh;
+    resVal = trunc(testVal);
+    if (baseVal != resVal) {
+        error++;
+    }
+
+    //Case 4 : Generic value -inf < val < -1 with deltaHigh.
+    baseVal = -aFOO[programCount - 1];
+    testVal = baseVal - deltaHigh;
+    resVal = trunc(testVal);
+    if (baseVal != resVal) {
+        error++;
+    }
+
+    //Case 5 : Value 0 < val < +0.5.
```

Comment:
Is there a difference between 0 < x < 1 and 1 < x +inf? This seems to be pretty much the same, I don't think it needs to be tested separately.

Reply:
Added comment explaining